Firestore/core/src/firebase/firestore/api/firestore.mm

```diff
@@ -176,6 +179,16 @@
   client_->DisableNetwork(std::move(callback));
 }
 
+std::unique_ptr<ListenerRegistration> Firestore::AddSnapshotsInSyncListener(
+    std::unique_ptr<core::EventListener<Empty>> listener) {
+  EnsureClientConfigured();
+  auto async_listener = AsyncEventListener<Empty>::Create(
+      client_->user_executor(), std::move(listener));
+  client_->AddSnapshotsInSyncListener(std::move(async_listener));
+  return absl::make_unique<SnapshotsInSyncListenerRegistration>(
+      std::move(client()), std::move(async_listener));
```

Comment:
`std::move(client())` is problematic:

  * The return type of `client()` is a const reference so `std::move()` has no effect and is misleading
  * You're mixing usage of `client_` (the field) and `client()` (the getter for that field) in the same method
  * Of course changing this to `std::move(client_)` would be disastrous (because it would give away the `client_` member, leaving the next operation in here to fail.

This should be just `client_` because we should be merely adding another reference to the client.